src/features/people/PeopleList/index.js

```diff
@@ -1,36 +1,44 @@
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { useParams } from "react-router-dom";
+import { ErrorPage } from "../../../common/ErrorPage";
+import Loader from "../../../common/Loader";
+import { Pagination } from "../../../common/Pagination";
+import { fetchPeople, selectFetchStatus, selectPeopleList } from "../peopleSlice";
 import { PersonTile } from "./PersonTile";
 import { Content, Title, Wrapper } from "./styled";
 
-export const PeopleList = () => {
-    return (
-        <Content>
-            <Title>Popular people</Title>
-            <Wrapper>
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-                <PersonTile name={"Liu Yifei"} />
-            </Wrapper>
-        </Content>
-    )
-};
+export const PeopleList = (props) => {
+    const dispatch = useDispatch();
+    const fetchStatus = useSelector(selectFetchStatus);
+    const people = useSelector(selectPeopleList);
+    const { page } = useParams();
+
+    useEffect(() => {
+        dispatch(fetchPeople(page));
+    }, [dispatch, page]);
+
+    switch (true) {
+        case (fetchStatus === "completed" && people !== undefined):
+            return (
+                <Content>
+                    <Title>Popular people</Title>
+                    <Wrapper>
+                        {people.map((person) =>
+                            <PersonTile key={person.id}
+                                name={person.name}
+                                portrait={ person.profile_path ? `https://image.tmdb.org/t/p/w185/${person.profile_path}` : null}
+                                id={person.id}
+                                // role={props.role}
```

Comment:
Because we WILL need it when we have a functioning movie page - role will need to be passed there to render full person tiles, should I already make it ready for use?

Reply:
We will not need it in PeopleList component. Role is already included in PersonTile props, but we will be passing this value in different component, right?

Reply:
Ah, now I get what you mean. True, we probably won't reuse PeopleList component in MoviePage - we'll probably just make a very similar component, we'll only reuse PersonTiles. Makes sense, I'll remove it